src/appengine/libs/query/datastore_query.py

```diff
@@ -53,12 +55,42 @@ def _combine(q1, q2):
   return result
 
 
+# TODO(ochang): Remove once bug is fixed.
+class QueryIteratorWrapper(object):
+  """Workaround for https://github.com/googleapis/python-ndb/issues/292."""
+
+  def __init__(self, it):
+    self._it = it
+    self._last_cursor = None
+
+  def __iter__(self):
+    return self
+
+  def __next__(self):
+    if not self._it.has_next():
+      try:
+        self._last_cursor = self._it.cursor_after()
+      except exceptions.BadArgumentError:
+        pass
+
+    return next(self._it)
+
+  next = __next__
```

Comment:
nite: move to start of class (not in middle of functions).

Reply:
This is a common pattern (see e.g. https://github.com/googleapis/python-ndb/blob/4ffedc7b5a2366be15dcd299052d8a46a748addd/google/cloud/ndb/_datastore_query.py#L468). Putting it before won't work either becausess __next__ won't be defined at that point. This is just to create an alias for python2/3 compatibility.